extensions/ql-vscode/src/run-queries.ts

```diff
@@ -411,10 +418,24 @@ export async function compileAndRunQueryAgainstDatabase(
   const query = new QueryInfo(qlProgram, db, packConfig.dbscheme, quickEvalPosition, metadata);
   await checkDbschemeCompatibility(cliServer, qs, query);
 
-  const errors = await query.compile(qs);
+  let errors;
+  try {
+    errors = await query.compile(qs);
+  } catch (e) {
+    if (e instanceof ResponseError && e.code == ErrorCodes.RequestCancelled) {
+      return createSyntheticResult(query, db, historyItemOptions, 'Query cancelled');
+    } else {
+      throw e;
+    }
+  }
 
   if (errors.length == 0) {
     const result = await query.run(qs);
```

Comment:
Do you want to take care of cancelled running queries right here too?

Reply:
Yes!

Reply:
Hmmm... `QueryResultType` is a namespace, which means that we can't pass it as a parameter type. Seems to me we should change it to an enum. Is there any reason why it's not?

Reply:
Done. Will include the change in my next push.